internal/gitinterface/blob.go

```diff
@@ -0,0 +1,38 @@
+package gitinterface
+
+import (
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+func ReadBlob(repo *git.Repository, blobID plumbing.Hash) (int, []byte, error) {
+	blob, err := repo.BlobObject(blobID)
+	if err != nil {
+		return -1, nil, err
+	}
+	contents := make([]byte, blob.Size)
+	reader, err := blob.Reader()
+	if err != nil {
+		return -1, nil, err
+	}
+	length, err := reader.Read(contents)
+	if err != nil {
+		return -1, nil, err
+	}
+	return length, contents, nil
```

Comment:
I'm not sure returning the length is very useful.

This should usually be == blob.Size / len(contents).
If by chance it's not, we're not returning the reader so callers won't be able to easily pick up where the last read left off.

If you want to guarantee everything is read, use `io.ReadAll`.

Reply:
Okay, switched to this and dismissed `length` altogether. This should also address https://github.com/adityasaky/gittuf/pull/33#discussion_r1143501940 but I'm still curious if `len` is used by the reader rather than capacity.

Reply:
I was wrong with the last comment - it uses len not cap. Sorry about that!

> Reader is the interface that wraps the basic Read method.
> 
> Read reads up to len(p) bytes into p.

Reply:
No worries, it's good to file this away for future :smile: